src/kibana/plugins/discover/index.html

```diff
@@ -38,7 +38,7 @@
     </div>
   </navbar>
 
-  <config config-template="configTemplate" config-object="opts" config-close="configClose" config-submit="fetch"></config>
+  <config config-template="configTemplate" config-object="opts" config-close="configClose" config-submit="opts.saveDataSource"></config>
```

Comment:
In master if you open the load panel, start to enter a filter, and then hit enter the search query will refresh. This change makes makes that bad behavior worse (now it saves the search source).

Lets just keep it simple and have the individual templates define their own forms and submit handlers by dropping the `config-submit` attribute and wrapping updating these templates in form elements:

https://github.com/spalger/kibana/blob/master/src/kibana/plugins/discover/controllers/discover.js#L79-L82